examples/grpc-kotlin/src/main/kotlin/example/armeria/grpc/kotlin/HelloServiceImpl.kt

```diff
@@ -57,69 +55,34 @@ class HelloServiceImpl : HelloServiceGrpc.HelloServiceImplBase() {
      *
      * @see lazyHello(HelloRequest, StreamObserver)
      */
-    override fun lotsOfReplies(request: HelloRequest, responseObserver: StreamObserver<HelloReply>) {
+    override fun lotsOfReplies(request: HelloRequest): Flow<HelloReply> {
         // You can also write this code without Reactor like 'lazyHello' example.
-        Flux.interval(Duration.ofSeconds(1))
-                .take(5)
-                .map { "Hello, ${request.name}! (sequence: ${it + 1})" }
-                // You can make your Flux/Mono publish the signals in the RequestContext-aware executor.
-                .publishOn(Schedulers.fromExecutor(ServiceRequestContext.current().contextAwareExecutor()))
-                .subscribe({
-                    // Confirm this callback is being executed on the RequestContext-aware executor.
-                    ServiceRequestContext.current()
-                    responseObserver.onNext(buildReply(it))
-                },
-                {
-                    // Confirm this callback is being executed on the RequestContext-aware executor.
-                    ServiceRequestContext.current()
-                    responseObserver.onError(it)
-                },
-                {
-                    // Confirm this callback is being executed on the RequestContext-aware executor.
-                    ServiceRequestContext.current()
-                    responseObserver.onCompleted()
-                })
+        return flow {
+            for (i in 1..5) {
+                delay(1000)
+                emit(buildReply("Hello, ${request.name}! (sequence: $i)")) // emit next value
```

Comment:
Can you check context between delay / emit?

Reply:
I guess preserving context is required - even better is if `context.eventLoop().inEventLoop();` but if thread is changing without losing context, we could probably fix that later.

Reply:
I thought they are running on the same thread. But `delay(1000)` changed the current thread to the coroutine default context thread. 
After some investigating, `flowOn(eventLoop().asCoroutineDispatcher())` make `delay()` and `emit()` run on Armeria event loop.

Reply:
This looks similar to the rxjava pattern we use, but doesn't seem to take advantage of kotlin coroutines fully. I guess it's because of `Dispatcher.Unconfined`. Did you get to try implementing a custom dispatcher for armeria? I found some tips here

Maybe this pattern works
` override val executor: Executor get() = RequestContext.current().contextAwareExecutor()`
https://github.com/Kotlin/kotlinx.coroutines/blob/d7de5f5ba66a8d005e5cbd03b18522112303fd54/kotlinx-coroutines-core/jvm/src/scheduling/Dispatcher.kt#L54

I guess we just need to extend this class

https://github.com/Kotlin/kotlinx.coroutines/blob/d7de5f5ba66a8d005e5cbd03b18522112303fd54/kotlinx-coroutines-core/jvm/src/Executors.kt#L19

Similar to how it's done here

https://github.com/Kotlin/kotlinx.coroutines/blob/d7de5f5ba66a8d005e5cbd03b18522112303fd54/kotlinx-coroutines-core/jvm/src/Executors.kt#L72

It looks sort of complicated, but I guess we need something to make the kotlin experience nice

Reply:
Thanks for sharing the useful links! Let me try to implement `ArmeriaDispatcher`. 👍

Reply:
I've tried to implement `ArmeriaDispatcher`. but faced a problem. 
When `ArmeriaDispatcher` is injected,
1. An Armeria event loop serves the coroutine stub directly.
2. `delay(..)` is handled by a coroutine default dispatcher.
3. when the delay is done, the dispatcher thread calls the `RequestContext.current().contextAwareExecutor()`

To solve this, we need to capture the executor. IMHO it seems to difficult in the current API.

I approached it in a different way. gRPC-Kotlin propagates gRPC context already.
https://github.com/grpc/grpc-kotlin/blob/2f3d86b661894d2047c98ca8082932dffc8378d1/stub/src/main/java/io/grpc/kotlin/ServerCalls.kt#L199

1. `ServiceRequestContext` is set to gRPC context before calling Coroutine Stub.
2. Extract `ServiceRequestContext` from gRPC context using ArmeriaContextElement.

If this workaround is reasonable, I would like to add `grpc-kotlin` module to Armeria.
/cc @anuraaga